stream mode，can 't detect answer is over

the code is bellow ：
```
for {
		response, err := stream.Recv()
		if errors.Is(err, io.EOF) {
			fmt.Println("Stream finished")
			return
		}

		if err != nil {
			fmt.Printf("Stream error: %v\n", err)
			return
		}

		fmt.Printf("%v\n", err)
		fmt.Printf("%T\n", response.Choices[0].Text)
		//fmt.Printf("%s", response.Choices[0].Text)
	}
```

this is echo ：
```
<nil>
string
<nil>
string
<nil>
string
<nil>
string
<nil>
string
<nil>
string
<nil>
string
<nil>
string
<nil>
string
<nil>
string
<nil>
string
<nil>
string
<nil>
string
<nil>
string
<nil>
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
main.main()
        /home/ly/Desktop/chatgpt/mydemo2/main.go:42 +0x545
```


then answer is over **err**  is not equal  **io.EOF**
```
		if errors.Is(err, io.EOF) {
			fmt.Println("Stream finished")
			return
		}
```

--- a/stream.go
+++ b/stream.go
@@ -7,6 +7,7 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"io"
 	"net/http"
 )
 
@@ -16,12 +17,18 @@ var (
 
 type CompletionStream struct {
 	emptyMessagesLimit uint
+	isFinished         bool
 
 	reader   *bufio.Reader
 	response *http.Response
 }
 
 func (stream *CompletionStream) Recv() (response CompletionResponse, err error) {
+	if stream.isFinished {
+		err = io.EOF
+		return
+	}
+
 	var emptyMessagesCount uint
 
 waitForData:
@@ -44,6 +51,8 @@ waitForData:
 
 	line = bytes.TrimPrefix(line, headerData)
 	if string(line) == "[DONE]" {
+		stream.isFinished = true
+		err = io.EOF
 		return
 	}
 

--- a/stream_test.go
+++ b/stream_test.go
@@ -5,6 +5,8 @@ import (
 	"github.com/sashabaranov/go-gpt3/internal/test"
 
 	"context"
+	"errors"
+	"io"
 	"net/http"
 	"net/http/httptest"
 	"testing"
@@ -75,7 +77,6 @@ func TestCreateCompletionStream(t *testing.T) {
 			Model:   "text-davinci-002",
 			Choices: []CompletionChoice{{Text: "response2", FinishReason: "max_tokens"}},
 		},
-		{},
 	}
 
 	for ix, expectedResponse := range expectedResponses {
@@ -86,6 +87,11 @@ func TestCreateCompletionStream(t *testing.T) {
 		if !compareResponses(expectedResponse, receivedResponse) {
 			t.Errorf("Stream response %v is %v, expected %v", ix, receivedResponse, expectedResponse)
 		}
+	}
+
+	_, streamErr := stream.Recv()
+	if !errors.Is(streamErr, io.EOF) {
+		t.Errorf("stream.Recv() did not return EOF in the end: %v", streamErr)
 	}
 }
 

